apps/rule-manager/app/service/SheetsRuleManager.scala

```diff
@@ -87,57 +79,54 @@ class SheetsRuleManager(credentialsJson: String, spreadsheetId: String) extends
             }
           }
 
-      if (errors.size != 0) {
+      if (errors.nonEmpty) {
         Left(errors)
       } else {
         Right(rules)
       }
     }
   }
 
-  private def getRuleFromRow(row: List[Any], index: Int): Try[Option[BaseRule]] = {
+  private def getRuleFromRow(row: List[Any], index: Int): Try[Option[DbRule]] = {
     try {
       val ruleType = row(PatternRuleCols.Type)
       val maybeIgnore = row.lift(PatternRuleCols.ShouldIgnore)
       val maybeId = row.lift(PatternRuleCols.Id).asInstanceOf[Option[String]]
       val rowNumber = index + 1
 
       (maybeId, maybeIgnore, ruleType) match {
-        case (_, Some("TRUE"), _) => Success(None)
-        case (None, _, _)         => Failure(new Exception(s"no id for rule (row: ${rowNumber})"))
-        case (Some(id), _, _) if id.length == 0 =>
+        case (None, _, _) => Failure(new Exception(s"no id for rule (row: ${rowNumber})"))
+        case (Some(id), _, _) if id.isEmpty =>
           Failure(new Exception(s"empty id for rule (row: ${rowNumber})"))
-        case (Some(id), _, "regex") =>
-          Success(
-            Some(
-              getRegexRule(
-                id,
-                row(PatternRuleCols.Pattern).asInstanceOf[String],
-                row(PatternRuleCols.Suggestion).asInstanceOf[String],
-                row(PatternRuleCols.Category).asInstanceOf[String],
-                row.lift(PatternRuleCols.Description).asInstanceOf[Option[String]]
-              )
-            )
-          )
-        case (Some(id), _, "lt") =>
+        case (Some(id), _, ruleType)
+            if !Set("regex", "lt", "lt_core").contains(ruleType.toString) =>
+          Failure(new Exception(s"Rule type ${ruleType} for rule with id ${id} not supported"))
+        case (Some(_), None, _) =>
+          Failure(new Exception(s"no Ignore column for rule (row: ${rowNumber})"))
+        case (Some(id), Some(ignore), ruleType) =>
           Success(
             Some(
-              getLTRuleXML(
-                id,
-                row(PatternRuleCols.Pattern).asInstanceOf[String],
-                row(PatternRuleCols.Category).asInstanceOf[String],
-                row(PatternRuleCols.Description).asInstanceOf[String]
+              DbRule.withUser(
+                id = None,
+                ruleType = ruleType.asInstanceOf[String],
```

Comment:
There's one problem I've found here – because the `ruleType` in our sheet doesn't perfectly match the `ruleType` [defined in our Db](https://github.com/guardian/typerighter/blob/83cfa3e362e609201bb8bd37da20b72877421fdb/apps/rule-manager/app/service/DbRuleManager.scala#L19), rule ingestion fails as soon as we encounter a difference (`regex` is the same, but `lt` -> `languageToolXML` and `lt_core` -> `languageToolCore` differ.)

To fix, let's map between the sheet and DB values here. (We could change them to match, but I think the DB values are clearer!)

If it's not clear how do that, v. happy to pair!

Reply:
Resolved in [f697f06](https://github.com/guardian/typerighter/pull/300/commits/f697f06c9bb5feebd299ea8e3aba6e3cff5b66b0)